src/Temporalio/Worker/Tuning/SlotReserveContext.cs

```diff
@@ -0,0 +1,57 @@
+using Temporalio.Bridge;
+
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// Context for reserving a slot from a <see cref="ICustomSlotSupplier"/>.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: Custom slot suppliers are currently experimental.
+    /// </remarks>
+    public class SlotReserveContext
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SlotReserveContext"/> class.
+        /// </summary>
+        /// <param name="ctx">The bridge version of the slot reserve context.</param>
+        internal SlotReserveContext(Temporalio.Bridge.Interop.SlotReserveCtx ctx)
+        {
+            this.SlotType = ctx.slot_type switch
```

Comment:
A bit surprised a switch expression works in older .NET (or maybe it doesn't, didn't check CI)